app/src/main/java/hous/release/android/presentation/hous/EditHousNameActivity.kt

```diff
@@ -37,19 +40,29 @@ class EditHousNameActivity :
     }
 
     private fun initBackBtnClickListener() {
-        binding.btnEditHousNameBack.setOnClickListener {
-            WarningDialogFragment().apply {
-                arguments = Bundle().apply {
-                    putSerializable(WARNING_TYPE, WarningType.WARNING_EDIT_HOUS_NAME)
-                    putParcelable(
-                        CONFIRM_ACTION,
-                        ConfirmClickListener(confirmAction = { finish() })
-                    )
-                }
-            }.show(supportFragmentManager, DIALOG_WARNING)
+        binding.btnEditHousNameBack.setOnClickListener { showWarningDialog() }
+
+        onBackPressedDispatcher.apply {
+            addCallback {
+                showWarningDialog()
+            }.also { callback ->
+                onBackPressedCallback = callback
+            }
```

Comment:
```suggestion
        onBackPressedDispatcher.addCallback {
                showWarningDialog()
            }
```
activity에서는 onBackPressedDispatcher를 굳이 멤버변수로 안만들어줘도 됩니당!!

Reply:
아까 물어보고 뺀다는걸,,, 까먹어버렸다! 고마워여 ㅎㅎ